[Mohamed Safwan] iP

mohamedsaf1 opened the pull request:
# DukeBruv

>"If you can dream it, you can do it" - Walt Disney ([source](https://www.brainyquote.com/quotes/walt_disney_130027))

*DukeBruv* enhances your daily lifestyle by efficiently keeping track of your tasks. It's
* user-friendly
* fast to use
* ad-free

All you need to do is,
1. download it from [here](https://github.com/mohamedsaf1/ip)
2. run it on your IDE/terminal
3. add your tasks
4. enjoy your very own task manager! :smile:

Features:
- [x] Managing tasks
- [x] Managing deadlines and timings
- [x] Aesthetic user interface (~~coming soon~~ try it out now!)
- [ ] Reminders (coming soon)

and it is **FREE**!

<hr />
If you Java programmer, you can use it to practice Java too. Here's the <code>main</code> method:

~~~ruby
public class Main {
   public static void main(String[] args) {
      Application.launch(MainApp.class, args);
   }
}

Commit:
Add Gradle support

mohamedsaf1 committed:
iP Level 6

mohamedsaf1 committed:
Level 6 with testing

mohamedsaf1 committed:
Merge branch 'master' into HEAD

mohamedsaf1 committed:
Level-6-edited

mohamedsaf1 committed:
Level-7

mohamedsaf1 committed:
Level-8

mohamedsaf1 committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Level-7

mohamedsaf1 committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Level-8

mohamedsaf1 committed:
Level-8

mohamedsaf1 committed:
A-Packages

mohamedsaf1 committed:
Merge branch 'add-gradle-support' of https://github.com/mohamedsaf1/ip

mohamedsaf1 committed:
A-Jar

mohamedsaf1 committed:
A-JavaDocs

mohamedsaf1 committed:
A-JavaDocs

mohamedsaf1 committed:
A-CodingStandard

mohamedsaf1 committed:
Level-9

mohamedsaf1 committed:
Merge branch 'A-JavaDoc'

* A-JavaDoc:
  A-JavaDocs
  A-JavaDocs

mohamedsaf1 committed:
Merge branch 'A-CodingStandard'

* A-CodingStandard:
  A-CodingStandard

mohamedsaf1 committed:
Merge branch 'Level-9'

* Level-9:
  Level-9

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java

peppapighs commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,51 @@
+     * Constructor for the Duke main class.
+     * @param filePath Location of the file that stores the data as a result of running the main class.
+     */
+    public Duke (String filePath) {
```
Should the space character between `Duke` and `(String filePath)` be removed? According to our coding standard:

> A method or **constructor** name stays attached to the open parenthesis `(` that follows it.

peppapighs commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,94 @@
+                    throw new DukeException("Ooops, the description of todo cannot be empty!");
+                }
+
+                String substringtd = s.replaceAll("todo ", "");
```
Instead of naming variable as `substringtd`, maybe consider renaming it to something likes `substringTd` or `todoSubstring`? According to our coding standard:

> Variable names must be in **camelCase**.

peppapighs commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,94 @@
+
+                String[] phrase = s.split("/by");
+                String substringdl1 = phrase[0].replaceAll("deadline", "");
+                String substringdl2 = phrase[1];
```
Same as above.

peppapighs commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,94 @@
+
+                String[] phrase = s.split("/at");
+                String substringdl1 = phrase[0].replaceAll("event", "");
+                String substringdl2 = phrase[1];
```
Same as above.

peppapighs reviewed (commented):
Overall, the code is structured nicely with OOP in mind. I notice that you include library files into the repo as well. Maybe consider modify `.gitignore` to exclude them. Some minor coding standard violations but other than that, everything looks good to me.

mohamedsaf1 committed:
GFMD update

angkl0 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,51 @@
+        Duke duke = new Duke("data/duke.txt");
+        duke.run();
+    }
+
```
Perhaps this empty space could be removed?

angkl0 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,97 @@
+     * Constructor to create Storage object.
+     * @param filePath Location of the file that stores the data as a result of running the main class.
+     */
+    public Storage (String filePath) {
```
Should it be Storage(String filePath)?

angkl0 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,51 @@
+
+    /**
+     * Method that overrides Java.toString() method to convert the Deadline
+     * task as a string.
```
There should be an empty line between description and parameters.

angkl0 reviewed (commented):
There could be more consistency in where you leave empty lines. eg there was no empty line between method header and try in getdate() method in deadline class but in load() method in storage class there is an empty line

mohamedsaf1 committed:
GFMD Update 2

mohamedsaf1 committed:
GFMD Final

yunruu reviewed (commented):
Good code! Maybe can try to make some parts more concise :)

mohamedsaf1 committed:
Level-10

mohamedsaf1 committed:
Merge branch 'branch-Level-10'

mohamedsaf1 committed:
Add assertions

The code assumes certain properties during runtime. This may cause bugs
in the program if the properties are not set correctly during runtime.

Add assertions to ensure that the assumptions made in the program during
runtime are valid.

Assert statements are added in the following classes which are where
crucial assumptions are made,
* Storage
* TaskList

mohamedsaf1 committed:
some minor changes.

mohamedsaf1 committed:
Code Quality: Lengthy code and unnecessary comments.

Readability of the code is important, so that code can be easily
understood in the future.

Let's extract some methods out to make the main methods shorter, and
use more specific names for variables.

mohamedsaf1 committed:
Merge pull request #2 from mohamedsaf1/branch-A-CodeQuality

Code Quality: Lengthy code and unnecessary comments.

mohamedsaf1 committed:
Merge branch 'master' of https://github.com/mohamedsaf1/ip

* 'master' of https://github.com/mohamedsaf1/ip:
  Code Quality: Lengthy code and unnecessary comments.

mohamedsaf1 committed:
Merge branch 'branch-A-Assertions'

* branch-A-Assertions:
  some minor changes.
  Add assertions

# Conflicts:
#	src/main/java/duke/Storage.java

mohamedsaf1 committed:
Add Help

Code does not include a help function for users. When in doubt, a help
function would be useful for users to continue using the app.

Let's add a help function and provide an avenue for users to clarify
their doubts on the app.

mohamedsaf1 committed:
Merge branch 'branch-C-Help'

* branch-C-Help:
  Add Help

mohamedsaf1 committed:
Add User Guide

mohamedsaf1 committed:
Add User Guide and a few minor changes.

User Guide is necessary in giving the users a guideline to use the
application, so that they have easy access to instructions and can
troubleshoot as well.

mohamedsaf1 committed:
Add link to User Guide.

mohamedsaf1 committed:
Fix bugs from peer review.
